### PR TITLE
Add min_y and max_y checks for Active Block Modifiers (ABM)

### PR DESCRIPTION
**Pull request by sfence**
Add min_y and max_y checks for Active Block Modifiers (ABM).

This check can be used by ABM to reduce CPU usage.

Example:
  vacuum mod has ABM for all airs nodes with vacuum node in the neighbourhood.
- This causes that for all air nodes neighbourhood has to be check.
- with this min_y, max_y, looking for the neighbourhood nodes is eliminated 

Some of minetest games (Hades Revisited for example) are using similar ABM calls like a vacuum with height limitation check, so it can help developers to better optimize ABM.

I believe this can be useful in multidimensional worlds where developers want to provide different behaviors in different worlds separated by Y limits for reducing server CPU usage.

## To do

This PR is a Ready for Review.

## How to test

Use vacuum mod from branch: [sfence_use_abm_min_max_y](https://github.com/sfence/vacuum/tree/sfence_use_abm_min_max_y) with enabled physics and measure abm time.
On slower CPUs (etc. slower than i7-4771, 3.5 GHz you should see something like: ```
2021-06-11 09:21:34: WARNING[Server]: active block modifiers took 201ms (processed 466 of 485 active blocks)
```
without min_y, max_y check and this message should disappear when minetest is compelled with min_y, max_y check-in ABM.

### CODE REVIEW

**Comment by nerzhul** [comment]
missing documentation. Else it sounds useful

**Comment by lhofhansl** [comment]
Good idea. Anything we can do to reduce the CPU load from ABMs is welcome.

**Comment by sfence** [comment]
> missing documentation. Else it sounds useful

I have added some lines about it to doc/lua_api.txt .

**Comment by AFCMS** [comment]
wow
MineClone2 would definitely benefit of this feature.

**Comment by rubenwardy** [comment]
This is missing a feature flag in `minetest.features`

**Comment by sfence** [comment]
> This is missing a feature flag in `minetest.features`

Should be fixed now.

**Review by SmallJoker — APPROVED** [review]

File: src/script/lua_api/l_env.h [hunk 1]
```diff
@@ -223,17 +223,21 @@ class LuaABM : public ActiveBlockModifier {
 	float m_trigger_interval;
 	u32 m_trigger_chance;
 	bool m_simple_catch_up;
+	int m_min_y;
+	int m_max_y;
```
Comment (sfan5) [inline src/script/lua_api/l_env.h:227]: should be `u16` typed

File: src/script/cpp_api/s_env.cpp [hunk 2]
```diff
@@ -150,13 +150,19 @@ void ScriptApiEnv::initializeEnvironment(ServerEnvironment *env)
 
 		bool simple_catch_up = true;
 		getboolfield(L, current_abm, "catch_up", simple_catch_up);
+		
+		int min_y = -31000;
```
Comment (sfan5) [inline src/script/cpp_api/s_env.cpp:154]: I know that the MT world ends earlier but this should simply be `INT16_MIN` to not make any assumptions

File: src/script/lua_api/l_env.h (hunk 1 above)
Comment (SmallJoker) [reply src/script/lua_api/l_env.h:227]: * `s16`

Comment (sfan5) [reply src/script/lua_api/l_env.h:227]: Right

**Comment by sfence** [comment]
Type s16 is used with INT16_MIN and INT16_MAX macros.

**Pull request merged**